manager/orchestrator/replicated/replicated_test.go

```diff
@@ -424,3 +425,133 @@ func TestReplicatedScaleDown(t *testing.T) {
 		}
 	})
 }
+
+func TestInitialization(t *testing.T) {
+	ctx := context.Background()
+	s := store.NewMemoryStore(nil)
+	assert.NotNil(t, s)
+	defer s.Close()
+
+	orchestrator := NewReplicatedOrchestrator(s)
+	defer orchestrator.Stop()
+
+	watch, cancel := state.Watch(s.WatchQueue())
+	defer cancel()
+
+	s1 := &api.Service{
+		ID: "id1",
+		Spec: api.ServiceSpec{
+			Annotations: api.Annotations{
+				Name: "name1",
+			},
+			Mode: &api.ServiceSpec_Replicated{
+				Replicated: &api.ReplicatedService{
+					Replicas: 2,
+				},
+			},
+		},
+	}
+
+	err := s.Update(func(tx store.Tx) error {
+		assert.NoError(t, store.CreateService(tx, s1))
+
+		nodes := []*api.Node{
+			{
+				ID: "node1",
+				Spec: api.NodeSpec{
+					Annotations: api.Annotations{
+						Name: "name1",
+					},
+					Availability: api.NodeAvailabilityActive,
+				},
+				Status: api.NodeStatus{
+					State: api.NodeStatus_READY,
+				},
+			},
+			{
+				ID: "node2",
+				Spec: api.NodeSpec{
+					Annotations: api.Annotations{
+						Name: "name2",
+					},
+					Availability: api.NodeAvailabilityActive,
+				},
+				Status: api.NodeStatus{
+					State: api.NodeStatus_READY,
+				},
+			},
+		}
+		for _, node := range nodes {
+			assert.NoError(t, store.CreateNode(tx, node))
+		}
+
+		// 2 rejected tasks are in store before orchestrator starts
+		tasks := []*api.Task{
+			{
+				ID:           "task1",
+				Slot:         1,
+				DesiredState: api.TaskStateReady,
+				Status: api.TaskStatus{
+					State: api.TaskStateRejected,
+				},
+				ServiceAnnotations: api.Annotations{
+					Name: "task1",
+				},
+				ServiceID: "id1",
+				NodeID:    "node1",
+			},
+			{
+				ID:           "task2",
+				Slot:         2,
+				DesiredState: api.TaskStateReady,
+				Status: api.TaskStatus{
+					State: api.TaskStateNew,
+				},
+				ServiceAnnotations: api.Annotations{
+					Name: "task2",
+				},
+				ServiceID: "id1",
+				NodeID:    "",
+			},
+		}
+		for _, task := range tasks {
+			assert.NoError(t, store.CreateTask(tx, task))
+		}
+
+		return nil
+	})
+	assert.NoError(t, err)
+
+	// Start the orchestrator.
+	go func() {
+		assert.NoError(t, orchestrator.Run(ctx))
+	}()
+
+	// wait for orchestration to finish
+	done := false
+	for !done {
+		select {
+		case <-watch:
+		case <-time.After(time.Second):
+			done = true
+		}
+	}
+
+	s.View(func(readTx store.ReadTx) {
+		tasks, err := store.FindTasks(readTx, store.ByServiceID("id1"))
+		require.NoError(t, err)
+		// after reconciliation, there should be 2 active tasks, others shutdown
+		var active []*api.Task
+		for _, task := range tasks {
+			switch {
+			case task.DesiredState <= api.TaskStateRunning:
+				active = append(active, task)
+			case task.DesiredState == api.TaskStateShutdown:
+				assert.True(t, task.Status.State == api.TaskStateRejected)
```

Comment:
Sorry, I'm still not seeing how the test confirms this. It looks like the `for !done {` loop is completely ignoring all the watch activity, and is equivalent to just `time.Sleep(time.Second)`.